AP_BLHeli: allow mixed type connection and stop motors if connection lost

IamPete1 opened the pull request:
This is two bug fixes for BLHeli suite connection following up from https://github.com/ArduPilot/ardupilot/pull/13058

If a mix of reversible and normal motors are connected it now reports 0 pwm allowing BLHeli suite to connect, and disables motor test functionality. This is because in this situation BLHeli suite is expecting all reversible ESC's so will output 1500 on all channels, this will be 50% throttle on normal motors. (I have not tested if motor test would work in this situation in anycase)

This also implements a 1 second connection time out when motors are active, currently if you pull the usb during a motor test it just leaves the motor at whatever pwm it was at, this now writes 1000 for normal and 1500 for reversible to the outputs on time out.

tridge commented on libraries/AP_BLHeli/AP_BLHeli.cpp:
```diff
@@ -1280,6 +1302,9 @@ void AP_BLHeli::update(void)
     debug("ESC: %u motors mask=0x%04x", num_motors, mask);
 
+    // check if we have a combination of reversable and normal
+    mixed_type = (mask != (mask & channel_reversible_mask.get())) & (channel_reversible_mask.get() != 0);
```
should be && ?

IamPete1 replied:
fixed

IamPete1 commented on libraries/AP_BLHeli/AP_BLHeli.cpp:
```diff
@@ -1280,6 +1302,9 @@ void AP_BLHeli::update(void)
     debug("ESC: %u motors mask=0x%04x", num_motors, mask);
 
+    // check if we have a combination of reversable and normal
+    mixed_type = (mask != (mask & channel_reversible_mask.get())) & (channel_reversible_mask.get() != 0);
```
second & should be &&

IamPete1 replied:
fixed

rmackay9 commented:
So we're ok for this to go in but just want to test it first.  once that's done it's OK to be merged

IamPete1 commented:
Tested this on a 'normal' copter, all esc's showed up and connected, motor test worked as expected.

IamPete1 commented:
Tested the test, rebased and changed the test debug messages to send texts so you always see the result of the test whether or not you have debug on, also now always reset the test param so you don't get bad channel printed endlessly if you try and test a motor you don't have
